packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -178,14 +232,47 @@ export const addAuthMiddlewares = (addConnectHandler) => {
     })
   }
   
+
+  // NB: You must also set the expressSessionSecret setting
+  const auth0ClientId = auth0ClientIdSetting.get();
+  const auth0OAuthSecret = auth0OAuthSecretSetting.get()
+  const auth0Domain = auth0DomainSetting.get()
+  if (auth0ClientId && auth0OAuthSecret && auth0Domain) {
+    passport.use(new Auth0Strategy(
+      {
+        clientID: auth0ClientId,
+        clientSecret: auth0OAuthSecret,
+        domain: auth0Domain,
+        callbackURL: `${getSiteUrl()}auth/auth0/callback`,
+      },
+      createAuth0UserHandler('services.auth0.id', profile => profile.id, async profile => {
+        delete profile._json
+        return {
+          email: profile.emails?.[0].value,
+          services: {
+            auth0: profile
+          },
+          username: await Utils.getUnusedSlugByCollectionName("Users", slugify(profile.displayName)),
+          displayName: profile.displayName,
```

Comment:
Is `profile.displayName` guaranteed to be set? This won't be added in the universal login, so it's not clear to me how this is going to be set immediately after login